File: M2K/pack1/ToaDo.java
package pack1;
import java.util.*;
import java.text.*;

public class ToaDo{
  private static DecimalFormat df;
  public double x,y;
  public ToaDo(ToaDo other){
    this(other.getX(),other.getY());
  }
  public ToaDo(double ax,double ay){
    setX(ax);
    setY(ay);
  }
  public double getX(){
    return x;
  }
  public double getY(){
    return y;
  }
  public void setX(double ax){
    x = ax;
  }
  public void setY(double ay){
    y = ay;
  }
  public double khoangCach(ToaDo other){
    return Math.sqrt( Math.pow((x - other.x),2) 
    + Math.pow((y - other.y),2) );
  }
  public String toDecimalString(){
    df = new DecimalFormat("0.000");
    return "(" + df.format(x) + "," + df.format(y) + ")";
  }
  public String toCircle(double banKinh){
    df = new DecimalFormat("0.000");
    return "(x-" + df.format(x) +")^2 +(y-" + df.format(y) + ")^2 = " + df.format(banKinh * banKinh) ;
  }
}

File: M2K/pack1/ToaDo.py
class ToaDo:
	def __init__(self, *args) -> None:
		self.x = self.y = 0
		if len(args) == 1:
			other = args[0]

			self.init(other.getX(), other.getY())
		
		else:
			ax, ay = args
			self.init(ax, ay)
	
	def init(self, ax, ay):
		self.setX(ax)
		self.setY(ay)
	
	def getX(self):
		return self.x

	def getY(self):
		return self.y
	
	def setX(self, ax):
		self.x = ax

	def setY(self, ay):
		self.y = ay

	def khoangCach(self, other):
		return (pow(self.x - other.x, 2) + pow(self.y - other.y, 2)) **.5
	
	def toDecimalString(self):
		return f'({self.x:.3f},{self.y:.3f})'

	def toCircle(self, R):
		return f'(x-{self.x:.3f})^2 +(y-{self.y:.3f})^2 = {R*R:.3f}'
	